src/components/TypeSelect/index.js

```diff
@@ -1,12 +1,12 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-function TypeSelect({getTepmeratureType, value}){
+function TypeSelect({getTemperatureType, value}){
   return(
     <>
       <form>
         <label> Choose a type </label>
-        <select onChange={getTepmeratureType} value={value}>
+        <select onChange={getTemperatureType} value={value}>
```

Comment:
function name staring with get means you expect return value, while assigning the function to onChange.
Change function name to handleTemperatureTypeChange